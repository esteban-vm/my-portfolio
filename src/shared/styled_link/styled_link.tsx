import type { LinkProps } from 'next/link'
import Link from 'next/link'

export default function StyledLink<T extends string>(props: LinkProps<T>) {
  return <Link {...props} legacyBehavior passHref />
}
