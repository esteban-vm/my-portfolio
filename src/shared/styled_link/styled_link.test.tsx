import { render, screen } from '@/tests'
import StyledLink from './styled_link'

describe(`${StyledLink.name}:`, () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <StyledLink href='/'>
        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
        <a>Test</a>
      </StyledLink>
    )

    const linkElem = screen.getByRole('link')
    expect(asFragment()).toMatchSnapshot()
    expect(linkElem).toBeInTheDocument()
    expect(linkElem).toHaveAttribute('href', '/')
    expect(linkElem).toHaveTextContent('Test')
  })
})
