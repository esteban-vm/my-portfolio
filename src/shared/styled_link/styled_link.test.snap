// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledLink: > should render correctly 1`] = `
<DocumentFragment>
  <a
    href="/"
  >
    Test
  </a>
</DocumentFragment>
`;
